projects/frontend/data-pipelines/gui/projects/ui/src/app/app.component.ts

```diff
@@ -24,33 +24,42 @@ export class AppComponent implements OnInit {
 
     constructor(
         private readonly appConfigService: AppConfigService,
-        private readonly oauthService: OAuthService,
+        @Optional() private readonly oauthService: OAuthService,
```

Comment:
Because `OAuthModule` is always imported in the `AppModule` (import happens in build time), service `OAuthService` will be always provided from `OAuthModule` (source code - https://github.com/manfredsteyer/angular-oauth2-oidc/blob/3ba7303b5a6fdbae5ca214764c9191637cd00126/projects/lib/src/provider.ts#L23) and Angular dependency injection will always inject that service into `AppComponent` so `@Optional` annotation won't change how it is resolved in runtime.
It is not wrong, but doesn't make any change in runtime, except for the unit tests purposes in the tests where service stub is not provided.

Reply:
That's true, the optional is a left-over from my prior attempt to have a conditional modules injection, and optimize the application context to only what's needed.

Reply:
Conditional module injection in a standard Angular application without some 3rd party module management, I think would be very hard to achieve, because Module Class decorator is resolved during build time and adding dynamic flags there is not an option.
You could check this article - https://mokkapps.de/blog/manually-lazy-load-modules-and-components-in-angular/ on a first check looks promising to load module in e.g. AppComponent